csrc/predicate_compute.cpp

```diff
@@ -558,7 +583,12 @@ Val* createElectSyncPredicate(kir::Predicate* pred) {
           NamedScalar::getParallelIndex(ParallelType::TIDx), zero);
     }
   }
-  return createElectSyncPredicate(/*use_first_warp=*/true, is_tma_store);
+
+  NVF_ERROR((!is_tma_store && !is_async_warp) || is_tma_store ^ is_async_warp);
```

Comment:
nit:
```
(!x && !y) || (x ^ y)
```
can be simplified as `x && y`

Reply:
| X | Y | NOT (X)|  NOT(Y) | NOT(X) AND NOT(Y) | XOR(x, y) |  (!X && !Y) OR (x ^ y) | X AND Y |
| - | - | - | - | - | - | - | - 
| F | F | T | T | T | F | T | F |
| F | T | T | F | F | T | T | F |
| T | F | F | T | F | T | T |  F |
| T | T | F | F | F | F | F | T |

I think you mean `!(X && Y)`

Reply:
Ahh, you are right